server/src/main/java/org/opensearch/identity/Subject.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.identity;
+
+import org.opensearch.identity.tokens.AuthToken;
+
+import java.security.Principal;
+
+/**
+ * An individual, process, or device that causes information to flow among objects or change to the system state.
+ *
+ * @opensearch.experimental
+ */
+public interface Subject {
+
+    /**
+     * Get the application-wide uniquely identifying principal
+     * */
+    Principal getPrincipal();
+
+    /**
+     * Login through an authentication token
+     * throws UnsupportedAuthenticationMethod
+     * throws InvalidAuthenticationToken
+     * throws SubjectNotFound
+     * throws SubjectDisabled
+     */
+    void login(final AuthToken token);
```

Comment:
Do you think that `login` may imply that a session is created? Maybe `authenticateRequest`?

Do you think it would make any sense to introduce `boolean isAuthenticated();` to this interface for a convenient function to check if the current subject is authenticated?

Reply:
Sure I can switch the name to `authenticateRequest` 

I will look at the code and see if the check method makes sense too!